src/internal/types.ts

```diff
@@ -0,0 +1,14 @@
+export type DeviceType = 'Handset' | 'Tablet' | 'Tv' | 'unknown';
+
+export type BatteryState = 'unknown' | 'unplugged' | 'charging' | 'full';
+
+export interface PowerState {
+  batteryLevel: number;
```

Comment:
I could be wrong (I have not double-checked) but the entries in PowerState might be optional? Like they may be missing and thus need a '?'

Reply:
This was copied from master, would need to check the native implementation on iOS to confirm this.  I know Android always returns these values, having recently implemented it.

Reply:
Looks like lowPowerMode is not always populated, but I suppose we could alter that to return false (our default boolean). Implementors choice though I think I have a slight preference for returning a default value then making library consumers have to special case that value https://github.com/react-native-community/react-native-device-info/blob/master/ios/RNDeviceInfo/RNDeviceInfo.m#L455

Reply:
maybe do one or the other as a separate PR so this can land now